### PR TITLE
Improve VRO end-to-end test

### PR DESCRIPTION
**Pull request by yoomlam**
## What was the problem?

VRO end-to-end test is minimal
- #681

## How does this fix it?

Improve VRO end-to-end test
- more logging to ease diagnostics
- show stacktrace in GH action when there's failure
- option to save PDF file
- add test for unauthenticated request

### CODE REVIEW

File: app/src/end2endTest/resources/logback.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<configuration>
+    <appender name="CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
```
Comment (yoomlam) [inline app/src/end2endTest/resources/logback.xml:2]: The `logback` Slf4j implementation is used (by default) for the end2end test.

File: app/src/end2endTest/java/gov/va/vro/end2end/util/TestSetup.java [hunk 2]
```diff
@@ -126,7 +130,7 @@ public String getContentDispositionFilename() {
    * @throws Exception any error to fail the test
    */
   public static TestSetup getInstance(String resourceDir) throws Exception {
-    TestSetup result = new TestSetup();
+    TestSetup result = new TestSetup(resourceDir);
```
Comment (yoomlam) [inline app/src/end2endTest/java/gov/va/vro/end2end/util/TestSetup.java:133]: Use the `resourceDir` as the name of the `TestSetup`. This name is used as the prefix for the PDF file that is optionally saved to disk.

File: app/src/end2endTest/java/gov/va/vro/end2end/VroV1Tests.java [hunk 3]
```diff
@@ -1,31 +1,64 @@
...
+    ContentDisposition actualCd = responseHeaders.getContentDisposition();
+    Assertions.assertEquals(setup.getContentDispositionFilename(), actualCd.getFilename());
+
+    checkPdfText(setup, response.getBody());
+  }
+
+  private void checkPdfText(TestSetup setup, byte[] pdf) throws Exception {
+    if (Boolean.parseBoolean(System.getenv("VRO_SAVE_PDF"))) {
+      savePdfFile(pdf, setup.getName() + "-" + setup.getContentDispositionFilename());
+    }
```
Comment (yoomlam) [inline app/src/end2endTest/java/gov/va/vro/end2end/VroV1Tests.java:61]: Save the PDF file if env var `VRO_SAVE_PDF` is set to true.

File: app/src/end2endTest/java/gov/va/vro/end2end/util/RestHelper.java [hunk 4]
```diff
@@ -35,63 +34,52 @@ private HttpHeaders buildHeaders() {
...
-    HttpHeaders headers = buildHeaders();
-    HttpEntity<String> requestEntity = new HttpEntity<>(req, headers);
+  public ResponseEntity<String> getAssessment(TestSetup setup) {
+    HttpEntity<String> requestEntity =
+        new HttpEntity<>(setup.getAssessmentRequest(), buildHeaders());
     String url = BASE_URL + ASSESSMENT_END_POINT;
+    log.info("POST to {} with request: {}", url, requestEntity);
     ResponseEntity<String> response = restTemplate.postForEntity(url, requestEntity, String.class);
-
-    Assertions.assertEquals(HttpStatus.CREATED, response.getStatusCode());
```
Comment (yoomlam) [inline app/src/end2endTest/java/gov/va/vro/end2end/util/RestHelper.java:48]: Move assertions to the test class.

File: app/build.gradle [hunk 5]
```diff
@@ -9,8 +9,17 @@ testSets {
   end2endTest
 }
 
-tasks.named("end2endTest") {
+tasks.withType(Test).named("end2endTest") {
   useJUnitPlatform()
+  testLogging {
+    // https://wjw465150.github.io/blog/Gradle/my_data/Gradle_Goodness/Show_Standard_Out_or_Error_Output_from_Tests.htm
+    showStandardStreams = true
+    events "skipped", "failed", "passed"
```
Comment (yoomlam) [inline app/build.gradle:17]: Be verbose, showing logs for all tests.

File: app/src/end2endTest/java/gov/va/vro/end2end/util/TestSetup.java [hunk 6]
```diff
@@ -12,26 +13,25 @@
...
           new AbstractMap.SimpleEntry<>("6602", "Asthma"));
-  private static final AtomicInteger claimSubmissionCounter = new AtomicInteger(7000);
+
+  private final String name;
 
   private String assessment;
   private String veteranInfo;
 
-  private String claimSubmissionId;
-
```
Comment (yoomlam) [inline app/src/end2endTest/java/gov/va/vro/end2end/util/TestSetup.java:34]: Using `claimSubmissionId` based on `claimSubmissionCounter` is fragile and dependent on the order in which each test runs. For test assertions, `claimSubmissionId` is expected to be equal to the one in the json files.

**Review by sethdarragile6 — APPROVED** [review]

**Review by au2 — APPROVED** [review]

**Pull request merged**